src/components/editor/editor.tsx

```diff
@@ -18,57 +18,67 @@
 // THE SOFTWARE.
 
-import React, {Component} from 'react';
-import PropTypes from 'prop-types';
+import React, {Component, ComponentType, CSSProperties, KeyboardEvent} from 'react';
 import styled from 'styled-components';
 import {Editor as Draw} from 'react-map-gl-draw';
 import window from 'global/window';
 import classnames from 'classnames';
 import get from 'lodash.get';
 import {createSelector} from 'reselect';
 
-import FeatureActionPanelFactory from './feature-action-panel';
+import FeatureActionPanelFactory, {FeatureActionPanelProps} from './feature-action-panel';
 import {FILTER_TYPES, EDITOR_MODES, EDITOR_AVAILABLE_LAYERS} from 'constants/default-settings';
 
 import {DEFAULT_RADIUS, getStyle as getFeatureStyle} from './feature-styles';
 import {getStyle as getEditHandleStyle, getEditHandleShape} from './handle-style';
 import KeyEvent from 'constants/keyevent';
+import {Layer} from 'layers';
+import {Datasets, Filter} from 'reducers';
+import {Feature} from '@nebula.gl/edit-modes';
+import {MjolnirEvent} from 'mjolnir.js';
 
 const StyledWrapper = styled.div`
-  cursor: ${props => (props.editor.mode === EDITOR_MODES.EDIT ? 'pointer' : 'crosshair')};
+  cursor: ${(props: {editor: {mode: string}}) =>
+    props.editor.mode === EDITOR_MODES.EDIT ? 'pointer' : 'crosshair'};
   position: relative;
 `;
 
-const editorLayerFilter = layer => EDITOR_AVAILABLE_LAYERS.includes(layer.type);
+// @ts-ignore
+const editorLayerFilter = (layer: Layer) => EDITOR_AVAILABLE_LAYERS.includes(layer.type);
 
 EditorFactory.deps = [FeatureActionPanelFactory];
 
-export default function EditorFactory(FeatureActionPanel) {
-  class Editor extends Component {
-    static propTypes = {
-      filters: PropTypes.arrayOf(PropTypes.object).isRequired,
-      layers: PropTypes.arrayOf(PropTypes.object).isRequired,
-      datasets: PropTypes.object.isRequired,
-      editor: PropTypes.object.isRequired,
-      layersToRender: PropTypes.object.isRequired,
-      onSelect: PropTypes.func.isRequired,
-      onUpdate: PropTypes.func.isRequired,
-      onDeleteFeature: PropTypes.func.isRequired,
-      onTogglePolygonFilter: PropTypes.func.isRequired,
-
-      index: PropTypes.number,
-      classnames: PropTypes.string,
-      clickRadius: PropTypes.number,
-      isEnabled: PropTypes.bool
-    };
+interface EditorProps {
+  filters: Filter[];
+  layers: Layer[];
+  datasets: Datasets;
+  editor: {selectedFeature: Feature; mode: string};
+  layersToRender: Record<string, Layer>;
+  index: number;
+  className: string; // ??
+  classnames: string;
+  clickRadius: number;
+  style: CSSProperties;
+  isEnabled: boolean;
+  onSelect: (f: Feature | null) => void;
+  onUpdate: (f: Feature[]) => void;
+  onDeleteFeature: (f: Feature) => void;
+  onTogglePolygonFilter: (l: Layer, f: Feature) => void;
+}
 
+export default function EditorFactory(
+  FeatureActionPanel: ComponentType<FeatureActionPanelProps>
```

Comment:
should this be `React.FC<FeatureActionPanelProps>`?

Reply:
fixed